### ISSUE TITLE
download_dsyms wait_for_dsym_processing is not working as expected when called from within another action.

### ISSUE THREAD

**Issue by MANassar** [fastlane/fastlane#16116]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
In my custom action, I am trying to call `download_dsyms` as follows:

`Actions::DownloadDsymsAction.run(wait_for_dsym_processing: true,	wait_timeout: 1800, app_identifier: ..., username: username, version: ..., build_number: ...)`

However, it does not wait for processing. It actually instantly fails with 

` No dSYM files found on App Store Connect - this usually happens when no recompiling has happened yet`

##### Command executed
`Actions::DownloadDsymsAction.run(wait_for_dsym_processing: true,	wait_timeout: 1800, app_identifier: ..., username: username, version: ..., build_number: ...)`

##### Complete output when running fastlane, including the stack trace and command used
` No dSYM files found on App Store Connect - this usually happens when no recompiling has happened yet`

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.3                                     |
| Ruby                        | 2.6.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.24.0                          |
| Installation Source         | ~/.rvm/gems/ruby-2.6.0/bin/fastlane         |
| Host                        | Mac OS X 10.15.3 (19D76)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.6.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.1b  26 Feb 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.3.1                                      |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

> Redacted

```
#!/bin/bash
# majorMinor=`cat version.properties | awk -F= '$1=="version"{print $2}'`
# /usr/libexec/PlistBuddy -c "Set :CFBundleShortVersionString $majorMinor" "$INFOPLIST_FILE"
# revisionNumber=$(date -u "+%y")
# buildNumber=$(date -u "+%m%d")
# /usr/libexec/PlistBuddy -c "Set :CFBundleVersion $majorMinor.$revisionNumber.$buildNumber" "$INFOPLIST_FILE"
```

</details>

<details><summary>`./fastlane/Appfile`</summary>

> Redacted 

</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.142.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                     | Version | Update-Status      |
| -------------------------- | ------- | ------------------ |
| fastlane-plugin-dynatrace  | 0.1.5   | 🚫 Update available |
| fastlane-plugin-versioning | 0.4.2   | ✅ Up-To-Date       |

<details><summary><b>Loaded gems</b></summary>

| Gem                        | Version      |
| -------------------------- | ------------ |
| did_you_mean               | 1.4.0        |
| CFPropertyList             | 3.0.2        |
| public_suffix              | 2.0.5        |
| addressable                | 2.7.0        |
| atomos                     | 0.1.3        |
| babosa                     | 1.0.3        |
| bundler                    | 2.1.4        |
| claide                     | 1.0.3        |
| colored                    | 1.2          |
| colored2                   | 3.1.2        |
| highline                   | 1.7.10       |
| commander-fastlane         | 4.4.6        |
| declarative                | 0.0.10       |
| declarative-option         | 0.1.0        |
| digest-crc                 | 0.4.1        |
| unf_ext                    | 0.0.7.6      |
| unf                        | 0.1.4        |
| domain_name                | 0.5.20190701 |
| dotenv                     | 2.7.5        |
| emoji_regex                | 1.0.1        |
| excon                      | 0.72.0       |
| multipart-post             | 2.0.0        |
| faraday                    | 0.17.3       |
| http-cookie                | 1.0.3        |
| faraday-cookie_jar         | 0.0.6        |
| faraday_middleware         | 0.13.1       |
| fastimage                  | 2.1.7        |
| gh_inspector               | 1.1.3        |
| jwt                        | 2.1.0        |
| memoist                    | 0.16.2       |
| multi_json                 | 1.14.1       |
| os                         | 1.0.1        |
| signet                     | 0.13.0       |
| googleauth                 | 0.11.0       |
| httpclient                 | 2.8.3        |
| mini_mime                  | 1.0.2        |
| uber                       | 0.1.0        |
| representable              | 3.0.4        |
| retriable                  | 3.1.2        |
| google-api-client          | 0.36.4       |
| google-cloud-env           | 1.3.0        |
| google-cloud-errors        | 1.0.0        |
| google-cloud-core          | 1.5.0        |
| google-cloud-storage       | 1.25.1       |
| json                       | 2.3.0        |
| mini_magick                | 4.10.1       |
| multi_xml                  | 0.6.0        |
| plist                      | 3.5.0        |
| rubyzip                    | 1.3.0        |
| security                   | 0.1.3        |
| naturally                  | 2.2.0        |
| simctl                     | 1.6.8        |
| slack-notifier             | 2.3.2        |
| terminal-notifier          | 2.0.0        |
| unicode-display_width      | 1.6.1        |
| terminal-table             | 1.8.0        |
| tty-screen                 | 0.7.1        |
| tty-cursor                 | 0.7.1        |
| tty-spinner                | 0.9.3        |
| word_wrap                  | 1.0.0        |
| nanaimo                    | 0.2.6        |
| xcodeproj                  | 1.15.0       |
| rouge                      | 2.0.7        |
| xcpretty                   | 0.3.0        |
| xcpretty-travis-formatter  | 1.0.0        |
| fastlane-plugin-dynatrace  | 0.1.5        |
| fastlane-plugin-versioning | 0.4.2        |
</details>

*generated on:* **2020-03-04**
</details>

So the question is, is this a bug? Or is this intended behavior that it does not wait on the loop in `DownloadDsymsAction`?

**Comment by MANassar** [comment]
Ping. Hello?

**Comment by kogutkarol** [comment]
I've had a similar problem and I was sure it was caused by some Fastlane update, but it appeared to be a problem with version number.

In logs I had something like this:
`[12:36:08]: Looking for dSYM files for '...' on platform appletvos v0 (69)
[12:36:08]: No dSYM files found on App Store Connect - this usually happens when no recompiling has happened yet`

I was ignoring the 'v0' for some time, but it was a key to solve my problem.

I used to get version number like this:
`build_version = get_info_plist_value(path: "...", key: "CFBundleShortVersionString")`
It seems that get_info_plist_value couldn't resolve $(MARKETING_VERSION), which has been put under 'CFBundleShortVersionString' by Xcode after some update.

Waiting for dsyms started to work again, when I've changed it to:

`get_version_number(target: "...")`

I don't know if this is the issue in your case, but maybe it will help someone ;).

Best regards,
Karol
